samples/goroutines3.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"fmt"
+	"runtime"
+	"time"
+)
+
+var nGoroutines int
+
+func memFunc() {
+	b := make([]int, 1024)
+	for i := 0; i < len(b); i++ {
+		b[0] = i
+		time.Sleep(1 * time.Microsecond)
+	}
+}
+
+func report() {
+	for {
+		n := runtime.NumGoroutine()
+		if n > 1 && nGoroutines != n {
+			nGoroutines = n
```

Comment:
I'm not sure what the point of `nGoroutines` is. It is set here but its never used for anything. I'm guessing this is left over from a more complicated example.
IMO simplify this function to just print the current number of goroutines.

Reply:
This is there to eliminate duplicate print statements. It now only prints if the number of goroutines have changed from the previous call. Else there are too many prints.